ribs/archives/_cvt_archive.py

```diff
@@ -164,16 +166,55 @@ def __init__(self,
         self._chunk_size = chunk_size
 
         if custom_centroids is None:
-            if not isinstance(samples, int):
-                # Validate shape of custom samples. These are ignored when
-                # `custom_centroids` is provided.
-                samples = np.asarray(samples, dtype=self.dtype)
-                if samples.shape[1] != self._measure_dim:
-                    raise ValueError(
-                        f"Samples has shape {samples.shape} but must be of "
-                        f"shape (n_samples, len(ranges)={self._measure_dim})")
-            self._samples = samples
-            self._centroids = None
+            if centroid_method == "kmeans":
+                if not isinstance(samples, int):
+                    # Validate shape of custom samples. These are ignored when
+                    # `custom_centroids` is provided.
+                    samples = np.asarray(samples, dtype=self.dtype)
+                    if samples.shape[1] != self._measure_dim:
+                        raise ValueError(
+                            f"Samples has shape {samples.shape} but must be of "
+                            f"shape (n_samples, len(ranges)="
+                            f"{self._measure_dim})")
+                self._samples = samples
+                self._centroids = None
+                if self._centroids is None:
```

Comment:
There are some problems with this logic. First, this if statement will always be activated since you set `self._centroids=None` right above it. Second, `samples` will always be overwritten. In the if statement for samples above, I think you should make an else that will generate the random samples with `uniform`. Then, you should always run the kmeans code down here.